src/Compilers/Core/Portable/Operations/CustomDataFlowAnalysis.cs

```diff
@@ -0,0 +1,402 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis
+{
+#if WORKSPACE
```

Comment:
This enables sharing the dataflow analysis between compiler and workspaces layer, while still retaining it as internal until we iterate on the appropriate API for eventually making it public.